magento/magento2
# MC-30171: Add to Cart Form wrong Form Key in FPC

### Description (*)

Fixes behavior of late form key update by js script. The product could be added to the cart with a cached form_key value that causes an exception.
### Related Pull Requests


### Fixed Issues (if relevant)

1. Fixes magento/magento2#13746

### Manual testing scenarios (*)

1. Magento installed
2. Simple product created
3. Full page cache flushed
4. Open product view page, note down form key
5. Open the same page in other browser or incognito window
6. Open page source
Expected result:
form_key is the same as in step 4. It is a cached value
7. Open normal page view
8. Open DevTools in the browser 
9. Enable Slow 3G throttling in the network tab
10. Refresh product page
11. Find form key element until page completely loaded
Expected result: 
Form key value differs from the value on steps 4 and 6
Correct form key value is loaded instantly when the element is rendered

### Questions or comments

Script logic in inline_form_key_provider.phtml can not be covered by any kind of automated tests. Just only manual testing.

### Contribution checklist (*)
